hyunjinjeon/app.js

```diff
@@ -0,0 +1,162 @@
+//Built-in
+const http =require ("http");
+
+//Third-party
+const express = require("express");
+const cors = require("cors");
+const morgan =  require("morgan");
+const dotenv = require("dotenv");
+const { DataSource } = require('typeorm'); 
+
+//Custom
+app = express();
+app.use(express.json());
+app.use(cors());
+app.use(morgan('combined'));
+dotenv.config()
+
+const appDataSource = new DataSource({
+    type: process.env.TYPEORM_CONNECTION,
+    host: process.env.TYPEORM_HOST,
+    port: process.env.TYPEORM_PORT,
+    username: process.env.TYPEORM_USERNAME,
+    password: process.env.TYPEORM_PASSWORD,
+    database: process.env.TYPEORM_DATABASE
+})
+
+//Error handling
+appDataSource.initialize()
+  .then(()=>{
+    console.log("Data Source has been initialized!")
+  });
+
+
+// 유저 회원 가입 엔드포인트
+app.post('/signup',async(req,res)=>{
+    const {name,email,password,age} = req.body
+    await appDataSource.query(
+        `INSERT INTO users (
+          user_name,
+          user_email,
+          user_password,
+          user_age
+       ) VALUES(?,?,?,?); 
+        `,[name,email,password,age]);
+
+    res.status(201).json({message:"userCreated"});
+})
+
+
+// 게시물등록 엔드포인트
+app.post('/posts',async(req,res)=>{
+    const {title,content} = req.body
+    await appDataSource.query(
+      `INSERT INTO posts (
+        title,
+        content
+     ) VALUES(?,?);
+    `,[title,content]);
+    res.status(201).json({message:"postCreated"});
+ })
+
+
+//게시물 전체조회 엔드포인트
+ app.get('/lookposts',async(req,res)=>{
```

Comment:
HTTP GET 메서드 자체의 의미를 보았을 때 특정 리소스를 조회하거나 찾는다는 의미가 있습니다.

따라서 RESTful API를 구현하는데 있어 엔드포인트에 동사형이 아닌 명사형으로 작성하도록 권장하고 있습니다.

```suggestion
 app.get('/posts', async(req, res)=>{
```